Playback/Core/RequestAllstarMatchRNG.asm

```diff
@@ -0,0 +1,55 @@
+################################################################################
+# Address: 801b5990
+################################################################################
+
+.include "Common/Common.s"
+.include "Playback/Playback.s"
+.include "TASBot.s"
+
+.set REG_MajorData, 30
+.set REG_Buffer, 29
+
+backup
+
+# alloc temp buffer
+  li r3, GET_ALLSTAR_RNG_BUFSIZE    # this is larger, using this to alloc the buffer
+  branchl r0, HSD_MemAlloc
+  mr REG_Buffer, r3
+
+RequestRNG:
+# cmd
+  li r3,CMD_REQ_ALLSTAR_RNG
+  stb r3,REQ_ALLSTAR_RNG_CMD_OFFSET(REG_Buffer)
+# difficulty
```

Comment:
I believe this is the 0-indexed match number, rather than difficulty! Just noting for reference